babel/src/test/java/org/apache/calcite/test/BabelParserTest.java

```diff
@@ -308,6 +308,23 @@ private void checkParseInfixCast(String sqlType) {
     sql(sql).ok(expected);
   }
 
+  @Test public void testCreateTablePermutedColumnLevelAttributes() {
```

Comment:
We should add test cases where only a subset of the attributes are explicitly specified (e.g. just "uppercase" or "not uppercase", but none of the others).

Reply:
With these new tests, we can verify that attributes which aren't specified are not written out during unparse.